### PR TITLE
NodeBalancer page in tabs and settings editable

### PR DESCRIPTION
**Pull request by eatonphil**
screen shot 2017-04-14 at 5 43 32 pm
screen shot 2017-04-14 at 5 43 45 pm

### CODE REVIEW

File: src/nodebalancers/nodebalancer/layouts/SettingsPage.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  async componentDidMount() {
+    const { dispatch } = this.props;
+    dispatch(setSource(__filename));
+  }
+
+  onSubmit = async () => {
+    const { dispatch, nodebalancer: { id, label: oldLabel } } = this.props;
+    const { group, label } = this.state;
+
+    await dispatch(dispatchOrStoreErrors.apply(this,
```
Comment (mparke) [inline src/nodebalancers/nodebalancer/layouts/SettingsPage.js]: why `apply` with an array of length `1`?

Comment (eatonphil) [inline src/nodebalancers/nodebalancer/layouts/SettingsPage.js]: I'm open to alternatives :D

Comment (mparke) [inline src/nodebalancers/nodebalancer/layouts/SettingsPage.js]: single arg `dispatchOrStoreErrors(() => nodebalancers.put({ group, label }, id))` should work the same with `(...apiCalls)`?

File: src/nodebalancers/nodebalancer/layouts/AddConfigPage.js [hunk 2]
```diff
@@ -69,50 +66,37 @@ export class AddConfigPage extends Component {
...
-              loading={loading}
-              errors={errors}
-              submitText="Add Configuration"
-            />
-          </Card>
+      <Card header={<CardHeader title="Add Configuration" />}>
+        <div>
+          <p>
+            Configure how your NodeBalancer listens for incoming traffic
+            and connects to backend nodes. <a href="#">Learn more.</a>
```
Comment (mparke) [inline src/nodebalancers/nodebalancer/layouts/AddConfigPage.js]: dead link? remove for now if we don't have a ref

File: src/nodebalancers/nodebalancer/layouts/SettingsPage.js (hunk 1 above)
Comment (eatonphil) [inline src/nodebalancers/nodebalancer/layouts/SettingsPage.js]: Ah, the point was to send it `this` without binding.

Comment (mparke) [inline src/nodebalancers/nodebalancer/layouts/SettingsPage.js]: then the returned function would need `apply(this`?, unless the async transpile is binding?

Comment (eatonphil) [inline src/nodebalancers/nodebalancer/layouts/SettingsPage.js]: `this` in the context of dispatchOrStoreErrors is the same `this` that is passed to the first argument of apply. Then dispatchOrStoreErrors returns a fat arrow function that captures that `this` inside of it.

**Comment by eatonphil** [comment]
Should be a much smaller PR after #1601

File: src/nodebalancers/nodebalancer/layouts/AddConfigPage.js (hunk 2 above)
Comment (eatonphil) [inline src/nodebalancers/nodebalancer/layouts/AddConfigPage.js]: done

File: src/nodebalancers/nodebalancer/layouts/SettingsPage.js (hunk 1 above)
Comment (mparke) [inline src/nodebalancers/nodebalancer/layouts/SettingsPage.js]: ah, yes, the fat arrow. okay cool

**Comment by mparke** [comment]
Unrelated to this pull, but found in testing: https://github.com/linode/manager/issues/1613

File: src/nodebalancers/nodebalancer/layouts/DashboardPage.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                  },
+                ]}
+                data={newConfigurations}
+                selectedMap={{}}
+                disableHeader
+              />
+            </ListBody>
+          </List>
+        </Card>
+        <Card header={<CardHeader title="Graph" />}>No stats are available.</Card>
```
Comment (mparke) [inline src/nodebalancers/nodebalancer/layouts/DashboardPage.js]: "stats" > "graphs" consistency

**Comment by mparke** [comment]
checked locally looks okay

File: src/nodebalancers/nodebalancer/layouts/DashboardPage.js (hunk 3 above)
Comment (eatonphil) [inline src/nodebalancers/nodebalancer/layouts/DashboardPage.js]: Done

**Pull request merged**